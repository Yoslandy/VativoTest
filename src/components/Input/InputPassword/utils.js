export const sizes = {
    small: "h-[40px] w-full",
    big: "h-[43px] w-full",
}
