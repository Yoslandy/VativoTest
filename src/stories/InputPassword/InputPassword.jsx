import React from "react"
import PropTypes from "prop-types"
import { Formik, Form } from "formik"
import InputPassword from "../../components/Input/InputPassword"

export default function CustomPassword(props) {
    const { name, ...rest } = props
    return (
        <Formik
            initialValues={{
                [name]: "",
            }}
        >
            <Form>
                <InputPassword name={name} {...rest} />
            </Form>
        </Formik>
    )
}

CustomPassword.propTypes = {
    name: PropTypes.string,
}

CustomPassword.defaultProps = {
    name: "password",
}
