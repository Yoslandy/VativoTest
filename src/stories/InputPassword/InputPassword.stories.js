import React from "react"

import "@fortawesome/fontawesome-free/css/all.min.css"
import "../../assets/styles/build.css"

import InputPassword from "./InputPassword"

export default {
    title: "Example/InputPassword",
    component: InputPassword,
    argTypes: {
        name: "Password",
    },
}

function Template(args) {
    return <InputPassword {...args} />
}

export const Default = Template.bind({})
Default.args = {
    name: "Password",
    placeholder: "Placeholder",
}
