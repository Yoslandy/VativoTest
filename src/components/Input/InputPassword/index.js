import React, { useState } from "react"
import { Field, ErrorMessage } from "formik"
import PropTypes from "prop-types"
import { sizes } from "./utils"
import Icon from "../../Icon"

export default function InputPassword(props) {
    const { name, placeholder, className, error, size, onFocus, label, hint } =
        props

    const [isPasswordType, setIsPasswordType] = useState(true)

    const handleActive = () => {
        onFocus()
    }

    const getErrorStyles = () =>
        error ? "input-error" : "border-[1px] border-primary-50"

    const renderError = () =>
        error && (
            <span className="formik-error-message">
                <ErrorMessage name={name} />
            </span>
        )

    return (
        <>
            <div className="flex flex-col">
                <label className="font-roboto text-[12px] text-neutral-gray-100 uppercase font-medium">
                    {label}
                </label>
                <div
                    className={`password-wrapper ${className} ${getErrorStyles()} ${
                        sizes[size]
                    }`}
                >
                    <Field
                        name={name}
                        placeholder={placeholder}
                        type={isPasswordType ? "password" : "text"}
                        className={`password-input ${
                            size === "big" ? "h-[41px]" : "h-[38px]"
                        }`}
                        onFocus={handleActive}
                    />
                    <div className="password-icon cursor-pointer">
                        <Icon
                            type={isPasswordType ? "eye" : "eye-slash"}
                            onClick={() => setIsPasswordType(!isPasswordType)}
                        />
                    </div>
                </div>
                {hint && (
                    <span className="font-roboto text-[14px] text-neutral-gray-100 text-center">
                        {hint}
                    </span>
                )}
                {renderError()}
            </div>
        </>
    )
}

InputPassword.propTypes = {
    name: PropTypes.string,
    placeholder: PropTypes.string,
    className: PropTypes.string,
    error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
    size: PropTypes.oneOf(["small", "big"]),
    onFocus: PropTypes.func,
    label: PropTypes.string,
    hint: PropTypes.string,
}

InputPassword.defaultProps = {
    name: "name",
    placeholder: "Jane",
    className: "",
    error: false,
    size: "big",
    onFocus: () => {},
    label: "",
    hint: "",
}
