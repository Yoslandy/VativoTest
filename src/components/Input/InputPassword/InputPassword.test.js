import React from "react"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import { Formik, Form } from "formik"
import InputPassword from "./index"

test("Renders Input Password form correctly", () => {
    const component = renderer.create(
        <MemoryRouter>
            <Formik
                initialValues={{
                    input: "",
                }}
            >
                <Form>
                    <InputPassword name="input" />
                </Form>
            </Formik>
        </MemoryRouter>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
