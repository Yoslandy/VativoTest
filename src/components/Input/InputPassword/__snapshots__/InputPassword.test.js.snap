// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Input Password form correctly 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="password-wrapper  border-[1px] border-primary-50 h-[43px] w-full"
  >
    <div
      className="password-with-asterisk"
    >
      <input
        className="password-input h-[41px] text-transparent"
        name="input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        placeholder="Jane"
        type="password"
        value=""
      />
      <span
        className="password-with-asterisk-dummy"
        id="input"
        style={
          Object {
            "display": "inline",
          }
        }
      />
    </div>
    <div
      className="password-icon"
    >
      <span
        className="material-icons "
        onClick={[Function]}
        role="button"
      >
        visibility
      </span>
    </div>
  </div>
  <span
    className="formik-error-message"
  />
</form>
`;
